source/jormungandr/jormungandr/scenarios/helper_classes/streetnetwork_path.py

```diff
@@ -95,6 +96,31 @@ def _direct_path_with_fp(self):
                 logging.getLogger(__name__).exception('')
                 return None
 
+    def _add_first_and_last_coord(self, dp):
+        if not dp or not dp.journeys or not dp.journeys[0].sections:
+            return
+
+        dp_coords = dp.journeys[0].sections[0].street_network.coordinates
+        if len(dp_coords) < 2:
+            return
+
+        # we are inserting the coord of the origin at the beginning of the geojson
+        coord = utils.get_pt_object_coord(self._orig_obj)
+        if coord != dp_coords[0]:
+            dp_coords.add(lon=coord.lon, lat=coord.lat)
+            # we cannot insert an element at the beginning of a list :(
+            # a little algo to move the last element to the beginning
+            tmp = type_pb2.GeographicalCoord()
+            for i in range(len(dp_coords)):
+                tmp.CopyFrom(dp_coords[i])
+                dp_coords[i].CopyFrom(dp_coords[-1])
+                dp_coords[-1].CopyFrom(tmp)
```

Comment:
How about something like this instead?

```python
dp_coords = [coord].extend(dp_coords)
```

I don't know how performant it should be though because your solution probably doesn't reallocate a list whereas mine probably does.

Reply:
unfortunately, `dp_coords` is a proto object, which means the objects can only be handled through its own API. An assignment that you suggest is not allowed and insertion at the begging is not allowed either :(